File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.go
/*
func trailingZeroes(n int) int {
    zeroes := 0
    for 0 < n {
        n /= 5
        zeroes += n
    }
    return zeroes
}
*/

func trailingZeroes(n int) int {
    divisor := 5
    zeroes := 0
    for divisor <= n {
        zeroes += n / divisor
        divisor *= 5
    }
    return zeroes
}

File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.py
class Solution:
    def trailingZeroes(self, n: int) -> int:
        if n == 0:
            return 0
        
        product = 1
        while n > 0:
            product *= n
            n -= 1
        
        str_product = str(product)
        zero_count = 0
        while str_product[-1] == "0":
            zero_count += 1
            str_product = str_product[:-1]
        
        return zero_count
        